oauth2/handler.go

```diff
@@ -689,6 +712,182 @@ func (h *Handler) getOidcUserInfo(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// swagger:route GET /oauth2/device/verify oAuth2 performOAuth2DeviceVerificationFlow
+//
+// # OAuth 2.0 Device Verification Endpoint
+//
+// This is the device user verification endpoint. The user is redirected here when trying to login using the device flow.
+//
+//	Consumes:
+//	- application/x-www-form-urlencoded
+//
+//	Schemes: http, https
+//
+//	Responses:
+//	  302: emptyResponse
+//	  default: errorOAuth2
+func (h *Handler) performOAuth2DeviceVerificationFlow(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	ctx := r.Context()
+
+	// When this endpoint is called with a valid consent_verifier (meaning that the login flow completed successfully)
+	// there are 3 writes happening to the database:
+	// - The flow is created
+	// - The device auth session is updated (user_code is marked as accepted)
+	// - The OpenID session is created
+	// If there were multiple flows created for the same user_code then we may end up with multiple flow objects
+	// persisted to the database, while only one of them was actually used to validate the user_code
+	// (see https://github.com/ory/hydra/pull/3851#discussion_r1843678761)
+	// TODO: We should wrap these queries in a transaction
```

Comment:
TODO is open here

Reply:
Yes, we discussed this with @zepatrik. We decided that there are no major issues with this and that it should be fixed later on